README.adoc

```diff
@@ -0,0 +1,402 @@
+
+
+
+
+## Plugins
+
+#### `fp` (https://github.com/jfmengels/eslint-plugin-fp[↗])
+Provides rules that help enforce functional programming.
+
+**Talks around FP**
+
+* https://www.infoq.com/presentations/Value-Values[The Value of Values] – Rich Hickey
+* https://www.infoq.com/presentations/Are-We-There-Yet-Rich-Hickey[Are we there yet?] – Rich Hickey
+* https://www.youtube.com/watch?v=DMtwq3QtddY[The Functional Final Frontier] – David Nolen
+* https://www.youtube.com/watch?v=mS264h8KGwk[Immutability, interactivity & JavaScript] – David Nolen
+
+**Writings around FP**
+
+* https://github.com/stoeffel/awesome-fp-js[Awesome FP JS] list
+* http://blog.wolksoftware.com/the-rise-of-functional-programming-and-the-death-of-angularjs[The rise of functional programming & the decline of Angular 2.0]
+
+#### `import` (https://github.com/benmosher/eslint-plugin-import[↗])
+Provides rules that help prevent import bugs and enforces style.
+
+#### `vue` (https://github.com/vuejs/eslint-plugin-vue[↗])
+Currently, the sole purpose of having this plugin is to parse https://vuejs.org/v2/guide/single-file-components.html[`.vue` files]. Also, it https://github.com/vuejs/eslint-plugin-vue/issues/1[does not support] eslint's `--fix` feature.
+
+## Extends
+
+#### Rules of AirBnB (https://github.com/airbnb/javascript[↗])
+
+We extend the AirBnB rules for historical reasons. Our configuration will continue to evolve and may not be based on it one day if we eventually disable or adjust too much of it via overrides.
+
+
+
+
+## Rules
+
+### `semi`
+
+Forbidden. +
+
+Refactoring is harder. +
+
+The act of changing/moving code  is made tedious by constantly baby sitting removal/addition of semicolons. For example: transitioning between `() => (...)` / `() => {...}`, moving the last `.then`/`.catch`/ `.foo` in a pipeline up or appending to it, etc.. +
+
+Reading and writing is harder. +
+
+JavaScript is a syntactically verbose language. Semicolons make it worse yet. Reading code with semicolons is objectively more fatiguing than code without given the reduction in visual noise. For writing, a day of coding with semicolons wears more on the fingers/hand than code without. +
+
+Further reading +
+
+* http://stackoverflow.com/questions/2846283/what-are-the-rules-for-javascripts-automatic-semicolon-insertion-asi[ASI rules]
+* http://blog.izs.me/post/2353458699/an-open-letter-to-javascript-leaders-regarding[An open letter to JavaScript leaders regarding Semicolons]
+* http://inimino.org/~inimino/blog/javascript_semicolons[JavaScript Semicolon Insertion; Everything you need to know]
+* http://mislav.net/2010/05/semicolons/[Semicolons in JavaScript are optional]
+
+**Fail**
+```js
+it("foobar", () => {
+  assert(1, foo(1));
+});
+```
+**Pass**
+```js
+it("foobar", () => {
+  assert(1, foo(1))
+})
+```
+
+### `quotes`
+
+Double. +
```

Comment:
Yeah that's fine. I think it's safe to assume they are ok to use but maybe we should mention it anyways?